### PR TITLE
En_Wood02

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Wood02/z_en_wood02.h [hunk 1]
```diff
@@ -5,11 +5,52 @@
...
+    /* 0x10 */ WOOD_BUSH_GREEN_LARGE_SPAWNED,
+    /* 0x11 */ WOOD_BUSH_BLACK_SMALL,
+    /* 0x12 */ WOOD_BUSH_BLACK_LARGE,
+    /* 0x13 */ WOOD_BUSH_BLACK_SMALL_SPAWNER,
+    /* 0x14 */ WOOD_BUSH_BLACK_SMALL_SPAWNED,
+    /* 0x15 */ WOOD_BUSH_BLACK_LARGE_SPAWNER,
+    /* 0x16 */ WOOD_BUSH_BLACK_LARGE_SPAWNED,
+    /* 0x17 */ WOOD_LEAF_GREEN,
+    /* 0x18 */ WOOD_LEAF_YELLOW,
+    /* 0x19 */ WOOD_19,
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Wood02/z_en_wood02.h:50]:
```suggestion
```
I checked, and type 19 does not actually exist. (Most of these are unused anyway, and are the same as OoT.)

File: src/overlays/actors/ovl_En_Wood02/z_en_wood02.c [hunk 2]
```diff
@@ -28,29 +50,466 @@ const ActorInit En_Wood02_InitVars = {
...
+        THIS->drawType = WOOD_DRAW_TREE_KAKARIKO_ADULT;
+    } else if (this->actor.params < WOOD_BUSH_BLACK_SMALL) {
+        THIS->drawType = WOOD_DRAW_BUSH_GREEN;
+    } else if (this->actor.params < WOOD_LEAF_YELLOW) {
+        THIS->drawType = WOOD_DRAW_4;
+    } else if (this->actor.params == WOOD_1A) {
+        THIS->drawType = WOOD_DRAW_TREE_CONICAL;
+    } else {
+        THIS->drawType = WOOD_DRAW_LEAF_YELLOW;
+    }
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Wood02/z_en_wood02.c:287]: Is there no other way?

File: src/overlays/actors/ovl_En_Wood02/z_en_wood02.c [hunk 3]
```diff
@@ -28,29 +50,466 @@ const ActorInit En_Wood02_InitVars = {
...
+                }
+            }
+
+            if ((thisx->child != NULL) && (this->unk_151 == 1)) {
+                thisx->child->home.rot.x = 1;
+                this->unk_151 = 0;
+            }
+
+            if (thisx->params == WOOD_1A) {
+                if (this->unk_144 != -1) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Wood02/z_en_wood02.c:422]: Can these ifs be combined?

File: src/overlays/actors/ovl_En_Wood02/z_en_wood02.c [hunk 4]
```diff
@@ -28,29 +50,466 @@ const ActorInit En_Wood02_InitVars = {
...
+            this->unk_146 = -0x15;
+            thisx->home.rot.y = 0;
+        }
+
+        if (thisx->xzDistToPlayer < 600.0f) {
+            Collider_UpdateCylinder(thisx, &this->collider);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+    } else if ((thisx->params < WOOD_LEAF_GREEN) || (thisx->params == WOOD_1A)) { // Bush
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Wood02/z_en_wood02.c:436]: This isn't just bushes any more, I'd say `// not leaves`

File: src/overlays/actors/ovl_En_Wood02/z_en_wood02.h [hunk 5]
```diff
@@ -5,11 +5,52 @@
...
+    /* 0x1A */ WOOD_1A
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Wood02/z_en_wood02.h:51]:
```suggestion
    /* 0x1A */ WOOD_TREE_SPECIAL = 0x1A
```
for now, based on behaviour (it's the only one that can spawn EnAni).

File: src/overlays/actors/ovl_En_Wood02/z_en_wood02.c (hunk 2 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Wood02/z_en_wood02.c:287]: Whatr do you mean?

Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Wood02/z_en_wood02.c:287]: Than using `THIS`, which is strongly discouraged.

File: src/overlays/actors/ovl_En_Wood02/z_en_wood02.c [hunk 6]
```diff
@@ -28,29 +50,466 @@ const ActorInit En_Wood02_InitVars = {
...
+                                      ((player->rideActor != NULL) && (sqrtf(thisx->xyzDistToPlayerSq) < 60.0f) &&
+                                       (player->rideActor->speedXZ != 0.0f)))) {
+            func_808C4458(this, globalCtx, &thisx->world.pos, 1);
+            this->unk_146 = -0x15;
+            Actor_PlaySfxAtPos(thisx, NA_SE_EV_TREE_SWING);
+        }
+    } else { // Leaves
+        this->unk_146++;
+        Math_ApproachF(&thisx->velocity.x, 0.0f, 1.0f, 0.049999997f);
+        Math_ApproachF(&thisx->velocity.z, 0.0f, 1.0f, 0.049999997f);
```
[the remark this answers is not in the document]
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Wood02/z_en_wood02.c:450]: Nope.

File: src/overlays/actors/ovl_En_Wood02/z_en_wood02.c (hunk 2 above)
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Wood02/z_en_wood02.c]: https://decomp.me/scratch/edcey
Seems to match?

File: src/overlays/actors/ovl_En_Wood02/z_en_wood02.c (hunk 6 above)
[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Wood02/z_en_wood02.c:448]: We found that `5.0f * 0.01f` should match in Discord.

File: src/overlays/actors/ovl_En_Wood02/z_en_wood02.c (hunk 2 above)
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Wood02/z_en_wood02.c]: This doesn't seem to have been addressed.

**Review by hensldm — APPROVED** [review]
LGTM, though Elliptic's comments still need to be addressed

**Review by EllipticEllipsis — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**